vector_getters.go

```diff
@@ -201,3 +201,49 @@ func (vec *vector) getSliceChild(offset uint64, length uint64) []any {
 	}
 	return slice
 }
+
+func (vec *vector) getUnion(rowIdx mapping.IdxT) any {
+	if vec.getNull(rowIdx) {
+		return Union{Tag: "", Value: nil}
+	}
+
+	// Get the tag vector (stored at index 0)
+	tagVector := mapping.StructVectorGetChild(vec.vec, 0)
+	if tagVector.Ptr == nil {
+		return Union{Tag: "", Value: nil}
+	}
+
+	// Get the tag value
+	tagData := mapping.VectorGetData(tagVector)
+	if tagData == nil {
+		return Union{Tag: "", Value: nil}
+	}
+	tags := (*[1 << 31]int8)(tagData)
+	tag := tags[rowIdx]
+
+	// Get the member vector
+	memberVector := mapping.StructVectorGetChild(vec.vec, mapping.IdxT(tag+1))
+	if memberVector.Ptr == nil {
+		return Union{Tag: "", Value: nil}
+	}
+
+	logicalType := mapping.VectorGetColumnType(vec.vec)
+	tagName := mapping.UnionTypeMemberName(logicalType, mapping.IdxT(tag))
+	mapping.DestroyLogicalType(&logicalType)
+
+	// Create a temporary vector with the member's type info and data
+	tempVec := vector{
+		vec:            memberVector,
+		dataPtr:        mapping.VectorGetData(memberVector),
+		maskPtr:        mapping.VectorGetValidity(memberVector),
+		vectorTypeInfo: vec.childVectors[tag].vectorTypeInfo,
+		getFn:          vec.childVectors[tag].getFn,
+		childVectors:   vec.childVectors[tag].childVectors,
+	}
+
+	value := tempVec.getFn(&tempVec, rowIdx)
```

Comment:
The struct's child vectors are already cached, we should access them directly via `tag`, which is a `int8t` index, no?

Then, I think we can rewrite all this to:
```Go
	child := &vec.childVectors[tag]
	value := child.getFn(child, rowIdx)
	return Union{
		Value: value,
		Tag:   child.indexDict[tag],
	}
```

`child.name` needs to be initialised during `init` - we need to cache the names for each tag (by index), maybe we can add another dictionary for that
```Go
type vectorTypeInfo struct {
	baseTypeInfo
	namesDict map[string]uint32
   indexDict map[uint32]string
}
```

Reply:
Updated the code. The tests are passing. Would be great if you could check again if the implementation makes sense. I have to admit I don't really understand what I am doing here and how data is represented internally in DuckDB. Thank you for bearing with me :pray:

Reply:
The `UNION` type is a logical type in DuckDB. That means its underlying 'physical type' has to be one of our existing physical types. We chose the `STRUCT` type - it has always as many children as there are tags in the union, plus one. That additional vector (the first child) is a `uint8` vector, and it contains the active tag of the row. Hope that helps. :)

Reply:
Thank you for explaining! And thanks for cleaning up the code around this :pray: